### PR TITLE
fix: add default port for protocol

### PR DESCRIPTION
**Pull request by olizilla**
multiaddrs are more explicit than urls. Default ports should always be included.

License: MIT
Signed-off-by: Oli Evans <oli@tableflip.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 [![Build Status](https://travis-ci.org/tableflip/uri-to-multiaddr.svg?branch=master)](https://travis-ci.org/tableflip/uri-to-multiaddr) [![dependencies Status](https://david-dm.org/tableflip/uri-to-multiaddr/status.svg)](https://david-dm.org/tableflip/uri-to-multiaddr) [![JavaScript Style Guide](https://img.shields.io/badge/code_style-standard-brightgreen.svg)](https://standardjs.com)
 
-> Convert a URI to a [Multiaddr](https://multiformats.io/multiaddr/): https://protocol.ai -> /dns4/protocol.ai/https
+> Convert a URI to a [Multiaddr](https://multiformats.io/multiaddr/): https://protocol.ai -> /dns4/protocol.ai/tcp/80/https
```
Comment (lidel) [inline README.md]: https → 443
```suggestion
> Convert a URI to a [Multiaddr](https://multiformats.io/multiaddr/): https://protocol.ai -> /dns4/protocol.ai/tcp/443/https
```

File: index.js [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
 /**
  * Convert a URI to a multiaddr
  *
- *        http://foobar.com => /dns4/foobar.com/https
- *  https://foobar.com:8080 => /dns4/foobar.com/tcp/8080/https
- *          ws://foobar.com => /dns4/foobar.com/ws
- *   https://127.0.0.1:8080 => /ip4/127.0.0.1/tcp/8080/https
- *        http://[::1]:8080 => /ip6/::1/tcp/8080/http
+ *        http://foobar.com => /dns4/foobar.com/tcp/80/http
+ *       https://foobar.com => /dns4/foobar.com/tcp/443/tls/http
```
Comment (lidel) [inline index.js]: I am not sure if discussion about `/tls/http` vs `/https` reached a conclusion yet:  https://github.com/multiformats/multiaddr/issues/63
Also, `/tls` is not defined in [multiaddr/protocols.csv](https://github.com/multiformats/multiaddr/blob/master/protocols.csv), so I think it may be _safer_ to go with `/https` for now (tests already do the right thing, we just need to fix this jsdoc):

```suggestion
 *       https://foobar.com => /dns4/foobar.com/tcp/443/https
```

File: index.js [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
+ *  https://foobar.com:5001 => /dns4/foobar.com/tcp/5001/tls/http
```
Comment (lidel) [inline index.js]: (same note as above)
```suggestion
 *  https://foobar.com:5001 => /dns4/foobar.com/tcp/5001/https
```

File: index.js [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
+ *   https://127.0.0.1:8080 => /ip4/127.0.0.1/tcp/8080/tls/http
```
Comment (lidel) [inline index.js]: (same note as above)
```suggestion
 *   https://127.0.0.1:8080 => /ip4/127.0.0.1/tcp/8080/https
```

File: index.js [hunk 5]
```diff
@@ -1,15 +1,14 @@
...
+ *        http://[::0]:8080 => /ip6/::0/tcp/8080/http
```
Comment (lidel) [inline index.js]: Why tho :thinking: For `ip4` (1 line above) we default to localhost, shouldn't we do the same here? (`::1`) 

I know this is a nit, but there is a huge difference if a beginner just copies&pastes `0.0.0.0:5001` vs `127.0.0.1:5001` without thinking, so I always try to avoid "listen on all interfaces" in examples :)

File: test.js [hunk 6]
```diff
@@ -3,8 +3,8 @@ const toMultiaddr = require('./')
 
 test('should convert URIs to multiaddrs', (t) => {
   const data = [
-    ['/ip4/127.0.0.1/http', 'http://127.0.0.1'],
-    ['/ip6/fc00::/http', 'http://[fc00::]'],
+    ['/ip4/127.0.0.1/tcp/80/http', 'http://127.0.0.1'],
```
Comment (lidel) [inline test.js]: Perhaps we should keep tests for both versions?

File: test.js [hunk 7]
```diff
@@ -37,6 +42,15 @@ test('should use the defaultDnsType where provided', (t) => {
...
-  t.throws(() => toMultiaddr('quic://'))
+test('should throw for on invalid url', (t) => {
+  t.throws(() => {
+    toMultiaddr('whoosh.fast')
+  }, /Invalid URL/)
+})
+
+test('should throw for unknown protocol', (t) => {
+  t.throws(() => {
+    // NOTE: `data` is a valid uri protocl but isn't a valid multiaddr protocol yet
```
Comment (lidel) [inline test.js]:
```suggestion
    // NOTE: `data` is a valid uri protocol but isn't a valid multiaddr protocol yet
```

This is a cool edge case! :)

File: index.js (hunk 5 above)
Comment (olizilla) [reply index.js]: 👍

File: test.js (hunk 6 above)
Comment (olizilla) [reply test.js]: You mean `ip4` and `ip6`?  Both tests have been kept, I just updated to include the port. Pesky diff.
